restapi/src/main/java/io/stargate/web/docsapi/service/DocsSchemaChecker.java

```diff
@@ -0,0 +1,66 @@
+package io.stargate.web.docsapi.service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.exception.ErrorCode;
+import io.stargate.web.docsapi.exception.ErrorCodeRuntimeException;
+import java.util.Objects;
+
+/**
+ * A helper class for checking that particular keyspace/table combinations exist and are valid
+ * document API collections.
+ */
+public class DocsSchemaChecker {
+  public static class KeyspaceAndTable {
+    private String keyspace;
+    private String table;
+
+    public KeyspaceAndTable(String keyspace, String table) {
+      this.keyspace = keyspace;
+      this.table = table;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      KeyspaceAndTable that = (KeyspaceAndTable) o;
+      return keyspace.equals(that.keyspace) && table.equals(that.table);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(keyspace, table);
+    }
+  }
+
+  private final Cache<KeyspaceAndTable, Boolean> validatedDocCollections =
+      Caffeine.newBuilder().maximumSize(10_000).build();
+
+  public void checkTableExists(String keyspace, String table, DocumentDB db) {
+    db.tableExists(keyspace, table);
+  }
+
+  public boolean isDocumentsTable(String keyspace, String table, DocumentDB db) {
+    KeyspaceAndTable keyspaceAndTable = new KeyspaceAndTable(keyspace, table);
+    Boolean valid = validatedDocCollections.getIfPresent(keyspaceAndTable);
+    if (valid == null) {
+      valid = db.isDocumentsTable(keyspace, table);
+      validatedDocCollections.put(keyspaceAndTable, valid);
+    }
+    return valid;
+  }
+
+  public void checkValidity(String keyspace, String table, DocumentDB db) {
+    checkTableExists(keyspace, table, db);
+    boolean isDocsTable = isDocumentsTable(keyspace, table, db);
```

Comment:
Is there a reason for calling `checkTableExists` outside of `isDocumentsTable`? 

If we check existence separately, would it make sense to also report it as different kind of error?

Reply:
checkTableExists throws an exception that marshalls to 404, but isDocumentsTable will produce a 400 (that's why I kept them separate, though maybe it should be clearer that `checkTableExists` is a void method that throws)

Reply:
I still have to do the cache invalidation on schema changes part btw - looking into that now

Reply:
Ah, I missed the 404 part, sorry :facepalm:

Reply:
ahh no let's not hit the `checkTableExists` every time, that was the reason why we wanted to have the cache at first place.. can we cache an enum as the value : `VALID`, `KEYSPACE_NOT_EXISTING`, `TABLE_NOT_EXISTING`, `TABLE_NOT_COLLECTION` and throw exception based on that??